stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -491,11 +492,17 @@ static void swift_taskGroup_destroyImpl(TaskGroup *group) {
 }
 
 void TaskGroupImpl::destroy() {
+  SWIFT_TASK_DEBUG_LOG("destroying task group = %p", this);
+
   // First, remove the group from the task and deallocate the record
   swift_task_removeStatusRecord(getTaskRecord());
 
-  // By the time we call destroy, all tasks inside the group must have been
-  // awaited on already; We handle this on the swift side.
+  // No need to drain our queue here, as by the time we call destroy,
+  // all tasks inside the group must have been awaited on already.
+  // This is done in Swift's withTaskGroup function explicitly.
+
+  // destroy the group's storage
+  this->~TaskGroupImpl();
```

Comment:
The fix™ 

Since the group is in place created in a manual allocation, we need to call the destructor.